src/wp-includes/blocks.php

```diff
@@ -1073,31 +1086,57 @@ function update_ignored_hooked_blocks_postmeta( $post ) {
 	}
 
 	/*
-	 * Skip meta generation when the post content is not a navigation block.
+	 * We only hook into filters for the `wp_navigation` and `wp_template_part` post types.
+	 * So if the post type is not set we can assume it's a `wp_template_part`.
+	 */
+	$post_type                   = isset( $post->post_type ) ? $post->post_type : 'wp_template_part';
```

Comment:
I _think_ we should infer the post type from the existing post object instead (if there is one).

If I'm not mistaken, the post type will be supplied as part of the `stdClass` object if the post object is to be newly created in the DB. This is true both for [template parts](https://github.com/WordPress/wordpress-develop/blob/6645c75e54c377d5adccc8e065562514f89a5427/src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php#L571-L593) and for [navigation post objects](https://github.com/WordPress/wordpress-develop/blob/6645c75e54c377d5adccc8e065562514f89a5427/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L1235-L1242). (This is in keeping with the general concept of the `stdClass` object: It contains all the information that needs to be changed -- or newly created -- when updating (or newly creating) the object in the DB. If the object already exists, it will already have a `post_type`.)